avm/res/data-factory/factory/main.bicep

```diff
@@ -79,195 +79,266 @@ param tags object?
 @description('Optional. Enable/Disable usage telemetry for module.')
 param enableTelemetry bool = true
 
-var formattedUserAssignedIdentities = reduce(map((managedIdentities.?userAssignedResourceIds ?? []), (id) => { '${id}': {} }), {}, (cur, next) => union(cur, next)) // Converts the flat array to an object like { '${id1}': {}, '${id2}': {} }
-
-var identity = !empty(managedIdentities) ? {
-  type: (managedIdentities.?systemAssigned ?? false) ? (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned') : (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'UserAssigned' : 'None')
-  userAssignedIdentities: !empty(formattedUserAssignedIdentities) ? formattedUserAssignedIdentities : null
-} : null
+var formattedUserAssignedIdentities = reduce(
```

Comment:
Hey @clintgrove, it would be great if we could enforce v2 formatting not only for this main.bicep template, but also for all other bicep files in this module, like, e.g., main.test.bicep and dependencies.bicep files.

To enforce v2 formatting, introduced by the latest bicep version by default, it's enough to:
- merge latest upstream main to this branch fork
- format all bicep files of this module to comply with v2. To do that it's enough to add a space anywhere in the file and save. The format on save repo setting will automatically format the file using v2

Reply:
Hi @eriqua thanks, noted. Will action this

Reply:
@eriqua  I think I have done what you asked above. Can you check and resolve this conversation if all is good please?

Reply:
@clintgrove well done on the `main.bicep` templates. The same should be done for any *.bicep file, including `dependencies.bicep` and `main.test.bicep` files in the tests/e2e folder

Reply:
Thank Erika  - please check the formatting now

Reply:
Just realised I did the databricks .bicep files instead of the datafactory ones!! THis shouldn't be a problem, but if its let me know

Reply:
Hey @clintgrove we can have both formatted with v2, but I'd suggest having that only if both validation pipelines are successful.
If that is not the case we should remove the changes related to the failing one.
Unfortunately, due to the auto formatting setting enabled, it may be difficult to revert the changes in this same branch. Since you are already working on another PR fix, it is probably easier to wait for that and merge this one next. What's your take on that?

Reply:
@eriqua  luckily the commit that I made databricks changes on was a commit that only had databricks changes, so I did  a git revert e75686405f1dba317c6141c294b9a72274bf3cd5 and it has reverted the changes of that commit. So we are all good to go